ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -2118,3 +2095,51 @@ def test033_Duplicate_main_order_and_cahange_materiel_type(self):
         self.assertEqual(suborder_data['Material Type'], material_type)
         self.assertEqual(suborder_data['Article Name'], article)
         self.assertEqual(suborder_data['Test Units'], test_unit)
+
+    def test034_duplicate_sub_order_table_with_add(self):
+        """
+        Orders: User can duplicate any suborder from the order form ( table with add )
+        LIMS-3738
+        :return:
+        """
+        # get random order
+        self.base_selenium.LOGGER.info('select random record')
+        orders, payload = self.orders_api.get_all_orders(limit=20)
+        data_before_duplicate_main_order = random.choice(orders['orders'])
+
+        #filter by this random order
+        order_no = data_before_duplicate_main_order['orderNo']
+        self.order_page.apply_filter_scenario(filter_element='orders:filter_order_no', filter_text=order_no,
+                                              field_type='text')
+        #get order data before duplicate
+        row = self.order_page.get_last_order_row()
+        self.orders_page.open_edit_page(row)
```

Comment:
you can use `https://github.com/modeso-open-source/1lims-automation/blob/b278dfc150ae4a54a0fdb25d1ea39f33c1081341/ui_testing/pages/orders_page.py#L15`
to open edit page directly without all above steps

Reply:
@ghada-37 what is this ?? I want to open the edit page for this specific order row last one I mean

Reply:
use it like this:
`get_order_edit_page_by_id(id=data_before_duplicate_main_order['id'])`

Reply:
@dianaeldemerdash Could u check this comment?

Reply:
done